crates/starknet/src/starknet/mod.rs

```diff
@@ -467,6 +576,23 @@ impl Starknet {
         add_invoke_transaction::add_invoke_transcation_v1(self, invoke_transaction)
     }
 
+    // save starknet transactions to file
+    pub fn dump_transactions(&self) {
+        // TODO: attach in OnTransaction mode or dump all??
+        match &self.config.dump_path {
+            Some(path) => {
+                let starknet_dump = Some(
+                    serde_json::to_string(&self.transactions)
+                        .expect("Failed to serialize starknet transactions"),
+                );
+                let encoded: Vec<u8> = bincode::serialize(&starknet_dump)
+                    .expect("Failed to encode starknet transactions");
+                fs::write(Path::new(&path), encoded).expect("Failed to save starknet transactions");
```

Comment:
This will override the whole file, need to use append option. Even if DumpOn::OnTransaction was activated all data in the file will be overridden

Reply:
I would propose considering creating `Dumper` entity that will have a queue of transaction hashes and when `dump_transactions` called exist queue is dumped

Reply:
I'll call you or dm about that.

Reply:
Sure

Reply:
I think that this can be implemented without a queue, I'll just attach a new transaction at the end of the file. I also moved the dump logic to another file 42f064dcf59fabf0427e179c8909cc3d6709b204.

Reply:
This can be addressed later once you will come back from vacation https://github.com/0xSpaceShard/starknet-devnet-rs/issues/197